File: Swift-C++-IR-Comparison/test/arithmetic-operators/Substraction/Swift/UnarySub.swift
@main
struct MyMain {

    static func main() {
        let b = -1
        let a = unaryMinus(x: b)
    }

    // The unary minus operator (-) inverts the current sign.
    // It performs the sign switch by calculating 0 - x
    static func unaryMinus(x: Int) -> Int {
        let tmp = -x
        return tmp
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-0/UnarySub.cpp
 
 struct MyMain {

public:
    static int unaryMinus(int x) {
        int a = x;
        int b = 0 - a;
        return b;
    }

    static void main() {
        int a = unaryMinus(x: -1);
    }
};

int main() {
    MyMain::main();

    return 0;
}